src/blob_file_iterator.cc

```diff
@@ -120,6 +136,10 @@ void BlobFileIterator::IterateForPrev(uint64_t offset) {
     status_ = decoder_.DecodeHeader(&header_buffer);
     if (!status_.ok()) return;
     total_length = kRecordHeaderSize + decoder_.GetRecordSize();
+    iterate_offset_ += total_length;
+    if (block_size_ != 0) {
```

Comment:
nit: `AdjustOffsetToNextBlockHead` already checks `block_size_` so the if check seems unnecessary. 

**Advance**OffsetToNextBlockHead may be a better name because we never expect the offset to go back. Personally I would rename it to **MaybeAdvance**OffsetToNextBlockHead but I don't know if that would be too long...